examples/chatgpt/gpt_actions_library/gpt_action_canvas.md

```diff
@@ -0,0 +1,584 @@
+
+1. [**General App Information**](#general-app-information) - Overview of Canvas LMS, its functionality, and the role of ChatGPT's Custom Actions to enhance educational experiences through AI integration.
+
+2. [**Authentication from ChatGPT to Canvas**](#authentication-from-chatgpt-to-canvas) - Explanation of authentication methods (OAuth and User Generated Access Tokens) for connecting ChatGPT to Canvas, with detailed instructions for setting up each option.
+
+3. [**Sample Use Case: Student Course Assistant**](#sample-use-case-student-course-assistant) - Detailed example of using ChatGPT to assist students with course navigation, exam preparation, and personalized feedback, including specific API calls and workflows.
+
+4. [**Other Use Cases for Consideration**](#other-use-cases-for-consideration) - Additional potential integrations using the Canvas API, such as classroom analytics and report generation.
+
+5. [**Congratulations**](#congratulations)  
+
+## General App Information
+
+Canvas is a widely-used Learning Management System (LMS) designed to support online learning and teaching. It offers a robust set of tools for course management, content delivery, assessments, and student collaboration. Through the [Canvas API](https://canvas.instructure.com/doc/api/all_resources.html), Canvas allows for extensive customization and integration with third-party applications, including AI-powered tools like ChatGPT. 
+
+ChatGPT’s Custom Actions with Canvas enable educators to leverage AI to enhance course content, automate tasks, and provide personalized learning journeys for students. Examples include virtual teaching assistants based on active courses, The capabilities are well-suited for pulling information in from Canvas to create an educational dialogue. ChatGPT with Custom Actions is not meant for automating the entire Canvas experience nor act as a replacement to many of its capabilities (like Assignment submission). 
+
+### Key Links
+
+https://canvas.instructure.com/doc/api/all_resources.html - API documentation used in the integration.
+
+https://canvas.instructure.com/doc/api/file.oauth.html#oauth2-flow - Authentication documentation and walkthrough.
+
+## Authentication from ChatGPT to Canvas
+
+- There are two options for authentication in Canvas: 1) OAuth and 2) User Generated Access Tokens.
+- For large-scale deployments, it is strongly recommended to use OAuth for Action Authentication. Access to Canvas’ Admin settings is required for OAuth in order to retrieve a Client ID and Client Secret.
+- If the user is considering a small-scale deployment or do not have access to Admin Settings, they may consider User Generated Access Tokens. Be aware that any request made by the action will be made using the token you generate, so Canvas will register all requests as your activity and use your permissions to complete them.
+
+### Implementing [OAuth Documentation](https://canvas.instructure.com/doc/api/file.oauth.html#oauth2-flow)
```

Comment:
@joecasson-openai changed the bullet points to title as I guess it refers more to how to implement, also made me figure out that we're missing a tutorial/recommendation on the OAuth implementation – any chances we can add this?

Reply:
Unfortunately I can't do it because I don't have access to a paid version of canvas. That's the only way I can become an admin to get the credentials needed for oauth.